Add Class Files

Wasif4life786 opened the pull request:
Very basic layout of class files

Wasif4life786 committed:
Initial commit for Maze Game

Lordimass reviewed (changes requested):
Many of the structure files you've created are class files, whereas on the diagrams by @Thumzu, they are interfaces, and abstract classes instead.

Wasif4life786 committed:
Added interfaces and abstract classes

Added interfaces and abstract classes, Had to change Event to an interface as interfaces can only extend other interfaces
